core/templates/components/forms/schema-based-editors/schema-based-int-editor.directive.ts

```diff
@@ -37,18 +40,28 @@ angular.module('oppia').directive('schemaBasedIntEditor', [
       template: require('./schema-based-int-editor.directive.html'),
       controllerAs: '$ctrl',
       controller: [
+        '$scope', '$timeout', 'FocusManagerService',
         'SchemaFormSubmittedService',
-        function(SchemaFormSubmittedService) {
+        function(
+            $scope, $timeout, FocusManagerService,
+            SchemaFormSubmittedService) {
           var ctrl = this;
+          var labelForFocus = $scope.labelForFocusTarget();
           ctrl.onKeypress = function(evt) {
             if (evt.keyCode === 13) {
               SchemaFormSubmittedService.onSubmittedSchemaBasedForm.emit();
             }
           };
+
           ctrl.$onInit = function() {
             if (ctrl.localValue === undefined) {
               ctrl.localValue = 0;
             }
+            // So that focus is applied after all the functions in
+            // main thread have executed.
+            $timeout(function() {
+              FocusManagerService.setFocusWithoutScroll(labelForFocus);
+            }, 50);
```

Comment:
@Rijuta-s Can you please elaborate what are we doing here? I think that this part of code is causing this behaviour https://github.com/oppia/oppia/issues/15918